addons/mrp/models/mrp_workorder.py

```diff
@@ -558,10 +558,17 @@ def _plan_workorder(self, replan=False):
         vals['leave_id'] = leave.id
         self.write(vals)
 
-    def _cal_cost(self):
+    def _cal_cost(self, date=False):
+        """Returns total cost of time spent on workorder.
+
+        :param date datetime: Only calculate for time_ids that ended before this date
+        """
         total = 0
         for wo in self:
-            duration = sum(wo.time_ids.mapped('duration'))
+            if date:
+                duration = sum(wo.time_ids.filtered(lambda t: t.date_end <= date).mapped('duration'))
+            else:
+                duration = sum(wo.time_ids.mapped('duration'))
             total += (duration / 60.0) * wo.workcenter_id.costs_hour
         return total
```

Comment:
```suggestion
        return self.company_id.currency_id.round(total)
```

Reply:
I believe it shouldn't be rounded here. We use this calculated cost in other places where it's sometimes aggregated + converted into other currencies